packages/utils/telemetry-utils/src/test/telemetryEventBatcher.spec.ts

```diff
@@ -56,12 +55,316 @@ describe("TelemetryEventBatcher", () => {
+		assert.strictEqual(loggerTwo.events.length, 1);
+
+		assert.strictEqual(resultOne.telemetryProperties.propertyOne, 1);
+		assert.strictEqual(resultOne.telemetryProperties.propertyTwo, 2);
+		assert.strictEqual(resultOne.telemetryProperties.propertyThree, 3);
+
+		assert.strictEqual(resultTwo.telemetryProperties.propertyOne, 4);
+		assert.strictEqual(resultTwo.telemetryProperties.propertyTwo, 5);
+		assert.strictEqual(resultTwo.telemetryProperties.propertyThree, 6);
+	});
+
+	it("correctly calculates data in different instances", () => {
+		const thresholdOne = 10;
+		const loggerOne = new TestLogger();
+		const batcherOne = new TelemetryEventBatcher<keyof TestTelemetryProperties>(
+			{ eventName: "testEventOne" },
+			loggerOne,
+			thresholdOne,
+		);
+		const spyOne = sinon.spy(loggerOne, "sendPerformanceEvent");
+
+		const thresholdTwo = 20;
+		const loggerTwo = new TestLogger();
+		const batcherTwo = new TelemetryEventBatcher<keyof TestTelemetryProperties>(
+			{ eventName: "testEventTwo" },
+			loggerTwo,
+			thresholdTwo,
+		);
+		const spyTwo = sinon.spy(loggerTwo, "sendPerformanceEvent");
+
+		for (let i = 1; i <= thresholdOne; i++) {
+			batcherOne.measure(() => ({
+				eventName: "testCall",
+				telemetryProperties: {
+					propertyOne: i + 10,
+					propertyTwo: i * 10,
+					propertyThree: i ** 2,
+				},
+			}));
+		}
+
+		for (let i = 1; i <= thresholdTwo; i++) {
+			batcherTwo.measure(() => ({
+				eventName: "testCall",
+				telemetryProperties: {
+					propertyOne: i + 100,
+					propertyTwo: i * 100,
+					propertyThree: i ** 3,
+				},
+			}));
+		}
+
+		const loggedEventOne = loggerOne.events[0];
+		const loggedEventTwo = loggerTwo.events[0];
+
+		assert.strictEqual(loggedEventOne.avgpropertyOne, 15.5);
+		assert.strictEqual(loggedEventOne.avgpropertyTwo, 55);
+		assert.strictEqual(loggedEventOne.avgpropertyThree, 38.5);
+		assert.strictEqual(loggedEventOne.maxpropertyOne, 20);
+		assert.strictEqual(loggedEventOne.maxpropertyTwo, 100);
+		assert.strictEqual(loggedEventOne.maxpropertyThree, 100);
+
+		assert.strictEqual(loggedEventTwo.avgpropertyOne, 110.5);
+		assert.strictEqual(loggedEventTwo.avgpropertyTwo, 1050);
+		assert.strictEqual(loggedEventTwo.avgpropertyThree, 2205);
+		assert.strictEqual(loggedEventTwo.maxpropertyOne, 120);
+		assert.strictEqual(loggedEventTwo.maxpropertyTwo, 2000);
+		assert.strictEqual(loggedEventTwo.maxpropertyThree, 8000);
+
+		spyOne.restore();
+		spyTwo.restore();
+	});
+
+	it("correctly calculates average, max values, and duration", () => {
+		const threshold = 10;
+		const eventBatcher = new TelemetryEventBatcher<keyof TestTelemetryProperties>(
+			{ eventName: "testEvent" },
+			logger,
+			threshold,
+		);
+
+		const clock = sinon.useFakeTimers();
+		const spy = sinon.spy(logger, "sendPerformanceEvent");
+
+		for (let i = 1; i <= threshold; i++) {
+			const startTime = i * 100;
+			const endTime = startTime + 50;
+			clock.tick(endTime - startTime);
```

Comment:
Are these necessary? Based on my cursory reading of the sinon fake timers APIs, calling `clock.tick(50)` inside measure should be enough to simulate "the function took 50 ms". Also, I think it'd be good if we make that ticking not constant, something like `clock.tick(50 + i)`, so then we can validate a more realistic scenario where the average duration that we expect in the loggedEvent isn't exactly equal to this value (so we know that an average was indeed computed and we didn't just log the latest value, for example).